### PR TITLE
Try ruby-slim

### PR DESCRIPTION
**Pull request by elalemanyo**
#10

shot2023-03-08 at 18 46 20@2x

### CODE REVIEW

**Comment by dhh** [comment]
Doesn't seem to build for me on ARM64? Complains of missing npm. Then after naming that explicitly, with an error with yarn.

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+RUN apt-get update -qq \
+  && DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends \
+    build-essential \
+    libvips \
+    gnupg2 \
+    curl \
+  && apt-get clean
+
+ARG PROJECT_NODE_MAJOR
+RUN curl -sL https://deb.nodesource.com/setup_19.x | bash -
```
[the remark this answers is not in the document]
Comment (elalemanyo) [reply Dockerfile]: @BilalBudhani you are right, I wanted to add some ARG's. I fix `PROJECT_NODE_MAJOR`, also add `RUBY_VERSION` and `DISTRO_NAME` btw.

**Comment by dhh** [comment]
Don't think build-arg'ing distro makes sense, given how different distros have different stuff installed and use different package managers.

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
+    gnupg2 \
+    curl \
+  && apt-get clean
+
+RUN curl -sL https://deb.nodesource.com/setup_$NODE_MAJOR.x | bash -
+RUN apt-get update -qq && DEBIAN_FRONTEND=noninteractive apt-get -yq dist-upgrade && \
+  DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends \
+    nodejs \
+    && npm install -g yarn \
+    && apt-get clean
```
Comment (viktorianer) [inline Dockerfile:20]: How about using multistage build to pull in binaries without having to install node, like it was proposed in https://github.com/rails/docked/pull/13?

Comment (elalemanyo) [reply Dockerfile:20]: I understand the advantages, but I think it's better the way we do it. But if everyone sees it better 🙂

**Comment by dhh** [comment]
If there's still interest here, we should follow the lead from the coming Rails Dockerfile. Just use regular slim: https://github.com/rails/rails/blob/main/railties/lib/rails/generators/rails/app/templates/Dockerfile.tt

**Comment by elalemanyo** [comment]
@dhh I have updated and now we use the same image as the coming Rails Dockerfile.

**Pull request merged**